### PR TITLE
[SOUNDS-8134] Update to look for `cacheStatus` variable from http-transport-cache instead of event listeners

### PR DESCRIPTION
**Pull request by craigharvi3**
**JIRA:** https://jira.dev.bbc.co.uk/browse/SOUNDS-8134

<hr />

## Summary
- From what I've been seeing during testing is that when we have requests going to multiple RMS endpoints (like we have in Radioplayer), because they are listening to events on the same event name (in our case, cache.rms.hit for example), I don't think we can guarantee that the listener that fires is for the actual request that triggered it - so when we try to build up a stats object to return to our RMS client, the stats may not be accurate.
- The hope is that https://github.com/bbc/http-transport-cache can return the cache status through the context for us to use, instead of having to rely on event listeners.

## Changelog
- Removed the event listeners binding
- Added check for `cacheStatus` in context
- If present, set the stats object to represent the status

## How to test it
Yarn link this to sounds-radioplayer and set the cacheStatus variable once/if https://github.com/bbc/http-transport-cache/issues/34 is pushed through.

### CODE REVIEW

File: src/stats.js [hunk 1]
```diff
@@ -155,12 +123,15 @@ export default async function stats(emitter, upstreamName, context, next) {
       };
     }
 
-    context.res.stats.attempts.push(attempt);
-
-    if (cacheAudit.length > 0) {
-      context.res.stats.cacheAudit.push(cacheAudit);
+    if (context.cacheStatus && Array.isArray(context.cacheStatus)) {
+      context.cacheStatus.forEach((cacheStatus) => {
+        attempt.cache[camelCase(cacheStatus, false, '_')] = true;
```
Comment (simonespa) [inline src/stats.js]: given `_` is the only separator we get, we could update the default in the `camelCase` util so we don't have to pass these extra parameters. What do you think?

**Review by simonespa — APPROVED** [review]
Looks good mate

File: src/stats.js (hunk 1 above)
Comment (craigharvi3) [reply src/stats.js]: Yep, makes sense. Updating.

Comment (craigharvi3) [reply src/stats.js]: Done d261519

**Pull request merged**